libyul/optimiser/SimplificationRules.h

```diff
@@ -85,14 +87,16 @@ enum class PatternKind
 class Pattern
 {
 public:
-	using Builtins = dev::eth::EVMBuiltins;
+	using Builtins = dev::eth::EVMBuiltins<Pattern>;
 	static constexpr size_t WordSize = 256;
```

Comment:
This is used in arithmetic in one of the rules in arithmetic resulting in MSVC looking for ``Pattern(size_t)`` as constructor... and apparently it doesn't seem to see that one resolved in the constructors below...
But we can't just add a ``Pattern(size_t)`` constructor, since that might conflict with the other ones...
It might be fine to have this be e.g. ``long unsigned`` instead and checking if that makes MSVC content?

Reply:
Actually I would prefer to only have Pattern-constructors for literal constants and force explicit Pattern construction for everything else, so it might be fine like that.

Reply:
Hm, yeah, ok... just ``size_t`` will match one of the "literal" constructors below on *most* platforms anyways...

Haha, we could use "user-defined literals" to actually restrict to literals only - like ``0_pattern`` (to be clear I'm not being serious with this).